modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/milker.dm

```diff
@@ -0,0 +1,319 @@
+
+/obj/item/milker/Initialize(mapload)
+	. = ..()
+	toggle_mode(null, TRUE)
+
+/obj/item/milker/attack(mob/living/M, mob/living/carbon/human/user)
+	if(!in_range(M, user))
+		to_chat(user, span_warning("[M] is too far away!"))
+		return FALSE
+	var/mob/living/carbon/being_milked = M
+
+	var/obj/item/organ/genital/breasts/breasts = being_milked.getorganslot(ORGAN_SLOT_BREASTS)
+	if(!breast_check(being_milked, user, breasts))
+		return FALSE
+
+	/// What's in our other hand? If it's a reagent container, we could try bottling it!
+	/// Holding a reagent container implies wanting to *use* the container, so if the container doesn't work, don't proceed!
+	var/obj/item/reagent_containers/milk_bottle = user.get_inactive_held_item()
+	if(istype(milk_bottle))
+		bottle_milk(being_milked, user, breasts, milk_bottle)
+	else // Or just drink it
+		drink_milk(being_milked, user, breasts)
+
+/// Set how vigorously you want to make the stuff come out
+/obj/item/milker/attack_self(mob/user)
+	toggle_mode(user, FALSE)
+
+/// Cycles through all the modes, looping around once it gets to the end
+/obj/item/milker/proc/toggle_mode(mob/user, milker_setup)
+	if(!milker_setup)
+		if(handle_mode++ > HANDLE_MODE_HIGHEST)
+			handle_mode = HANDLE_MODE_LOWEST
+	switch(handle_mode)
+		if(HANDLE_LIGHT)
+			if(user)
+				user.visible_message(
+				message = span_purple("[user] relaxes [user.p_their()] jaw and softens [user.p_their()] milking hand, appearing to opt for a gentler approach."),
+				self_message = span_purple("You relax your jaw and soften your milking hand, opting for a gentler approach."),
+				blind_message = span_purple("You hear a relaxed rustle."),
+				vision_distance = 1)
+			desc = "[initial(desc)]" + "\n[span_purple("It looks somewhat gentle.")]"
+			pleasure_amount = PLEASURE_LIGHT
+			arousal_amount = AROUSAL_LIGHT
+			pain_amount = PAIN_LIGHT
+			squirt_volume = VOLUME_LIGHT
+		if(HANDLE_STRONG)
+			if(user)
+				user.visible_message(
+				message = span_purple("[user] tightens [user.p_their()] jaw and clenches [user.p_their()] milking hand, appearing to opt for a more aggressive approach."),
+				self_message = span_purple("You tighten your jaw and clench your milking hand, opting for a more aggressive approach."),
+				blind_message = span_purple("You hear a vigorous rustle."),
+				vision_distance = 1)
+			desc = "[initial(desc)]" + "\n[span_purple("It looks rather aggressive!")]"
+			pleasure_amount = PLEASURE_STRONG
+			arousal_amount = AROUSAL_STRONG
+			pain_amount = PAIN_STRONG
+			squirt_volume = VOLUME_STRONG
+
+/// Checks if the breasts are present, exposed, lactating, in range, and containing some kind of fluid. Returns a message if not!
+/obj/item/milker/proc/breast_check(mob/living/carbon/being_milked, mob/living/carbon/human/milker, obj/item/organ/genital/breasts/breasts)
+	var/self_suckle = (being_milked == milker) // Grammarize
+	if(!being_milked.client?.prefs?.read_preference(/datum/preference/toggle/master_erp_preferences))
```

Comment:
```suggestion
	if(!being_milked.client || !being_milk.client.prefs.read_preference(/datum/preference/toggle/master_erp_preferences))
```
Basically just apply this to the rest of the PR I don't wanna repeatedly review the same thing